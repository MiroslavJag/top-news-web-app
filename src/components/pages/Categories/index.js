import React from 'react'
import styled from 'styled-components'

const Content = styled.h1`
  margin: 0;
`

const Categories = () => {
  return (
    <>
      <Content>I am Categories</Content>
    </>
  )
}

export default Categories
