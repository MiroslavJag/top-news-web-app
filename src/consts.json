{
 "ERRORS": {
  "API_ERROR": "API ERROR"
 },
 "LANG_CODES": {
  "GB": "gb",
  "US": "us"
 },
 "TOP_NEWS_LANG": "TopNewsLang",
 "CATEGORY_NAME": {
  "BUSINESS": "business",
  "ENTERTAINMENT": "entertainment",
  "GENERAL": "general",
  "HEALTH": "health",
  "SCIENCE": "science",
  "SPORTS": "sports",
  "TECHNOLOGY": "technology"
 }
}