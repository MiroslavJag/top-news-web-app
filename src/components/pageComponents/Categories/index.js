import React, {useState, useEffect} from 'react'
import styled from 'styled-components'
import * as actions from 'store/actions'
import {connect} from 'react-redux'
import {useTranslation} from 'react-i18next'
import NoImage from 'static/images/no-image-thumbnail.jpg'
import {LANG_CODES} from 'consts.json'
import {Separator, NewsListTitleLang} from 'components/pageComponents/Home'
import Slider from 'components/common/Slider'
import {CATEGORY_NAME} from 'consts.json'
import SingleNews from 'components/common/SingleNews'

const CategoriesPageWrapper = styled.div`
  text-align: left;
`
const CategoryPageTitleWrapper = styled.div`
  display: flex;
  flex-direction: row;
  border-bottom: 1px solid lightgrey;
`
const CategoryPageTitle = styled.h1`
  margin: 0 0 15px 0;
  text-align: left;
`
const CategoryContent = styled.div`
  margin: 20px 0 0 20px;
`

type Props = {
  langCode: String,
  setLanguageDisabled: Function,
}

const Categories = ({langCode, setLanguage, setLanguageDisabled}): Props => {
  const [singleNewsData, setSingleNewsData] = useState(null)
  const {t} = useTranslation()

  useEffect(() => {
    if (singleNewsData !== null) {
      setLanguageDisabled(true)
    }
  }, [singleNewsData])


  const renderTopNewsCategories = (
    <CategoriesPageWrapper>
      <CategoryPageTitleWrapper>
        <CategoryPageTitle>{t('TOP_NEWS_TITLE')}</CategoryPageTitle>
        <Separator />
        <NewsListTitleLang>{langCode}</NewsListTitleLang>
        <Separator />
        <CategoryPageTitle>{t('CATEGORIES')}</CategoryPageTitle>
      </CategoryPageTitleWrapper>
      <CategoryContent>
        <Slider category={CATEGORY_NAME.BUSINESS} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.ENTERTAINMENT} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.GENERAL} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.HEALTH} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.SCIENCE} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.SPORTS} setSingleNewsData={setSingleNewsData} />
        <Slider category={CATEGORY_NAME.TECHNOLOGY} setSingleNewsData={setSingleNewsData} />
      </CategoryContent>
    </CategoriesPageWrapper>
  )

  return (
    <>
      {singleNewsData !== null ? (
        <SingleNews singleNewsData={singleNewsData} setSingleNewsData={setSingleNewsData} />
      ) : (
        renderTopNewsCategories
      )}
    </>
  )
}
const mapStateToProps = (state) => {
  return {
    langCode: state.langCode,
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    setLanguageDisabled: (value) => dispatch(actions.setLanguageDisabled(value)),
    setLanguage: (langCode) => dispatch(actions.setLanguage(langCode)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Categories)
