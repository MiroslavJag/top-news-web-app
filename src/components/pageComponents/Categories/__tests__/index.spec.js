import React from 'react'
import {mount} from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Categories from 'components/pageComponents/Categories'
import {BrowserRouter} from 'react-router-dom'
import {Provider} from 'react-redux'
import configureStore from 'redux-mock-store'
import {LANG_CODES} from 'consts.json'

const mockStore = configureStore([])

describe('<Categories />', () => {
  let store
  let component

  beforeEach(() => {
    store = mockStore({
      langCode: LANG_CODES.GB,
      langDisabled: false,
    })

    component = (
      <Provider store={store}>
        <BrowserRouter>
          <Categories t={(k) => k} />
        </BrowserRouter>
      </Provider>
    )
  })

  it('match snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should render correctly with all needed components', () => {
    const wrapper = mount(component)
    expect(wrapper.find('Categories__CategoryContent')).toHaveLength(1)
    expect(wrapper.find('Slider__SliderWrapper')).toHaveLength(7)
  })
})
