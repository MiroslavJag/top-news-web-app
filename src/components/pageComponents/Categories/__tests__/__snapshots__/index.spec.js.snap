// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> match snapshot 1`] = `
.c4 {
  margin: 0 0 10px 0;
  text-align: left;
}

.c5 {
  text-transform: uppercase;
}

.c3 {
  margin: 0px 15px 0;
  height: 35px;
  width: 1px;
  background-color: lightgrey;
}

.c12 {
  max-with: 1340px;
  width: 100%;
  display: grid;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c7 > a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: darkslategrey;
}

.c7 > a:hover {
  color: grey;
}

.c9 {
  margin-right: 20px;
  text-transform: capitalize;
}

.c8 {
  height: 10px;
  width: 10px;
  margin-right: 20px;
  background-color: lightgray;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 0;
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-timing-function: cubic-bezier(0,1,1,1);
  transition-timing-function: cubic-bezier(0,1,1,1);
}

.c10 {
  height: 20px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  cursor: pointer;
}

.c0 {
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  border-bottom: 1px solid lightgrey;
}

.c2 {
  margin: 0 0 15px 0;
  text-align: left;
}

.c6 {
  margin: 20px 0 0 20px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      TOP_NEWS_TITLE
    </h1>
    <div
      className="c3"
    />
    <h1
      className="c4 c5"
    >
      gb
    </h1>
    <div
      className="c3"
    />
    <h1
      className="c2"
    >
      CATEGORIES
    </h1>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/business"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          BUSINESS
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/entertainment"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          ENTERTAINMENT
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/general"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          GENERAL
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/health"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          HEALTH
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/science"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          SCIENCE
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/sports"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          SPORTS
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
      />
      <a
        href="/categories/technology"
        onClick={[Function]}
      >
        <h2
          className="c9"
        >
          TECHNOLOGY
        </h2>
      </a>
      <img
        className="c10"
        onClick={[Function]}
        src="down-arrow.svg"
      />
    </div>
    <div
      className="c11"
    >
      <div
        className="c12"
      />
    </div>
  </div>
</div>
`;
